### PR TITLE
Replace RWOperation with map<string, string>

### LINKED ISSUE #1368: RWOperation.h replace with map<std::string, std::string>
The RWOperation class should be replaced with a simple map<std::string, std::string>. There is no need to introduce a new class for this simple map.

### PR DESCRIPTION
**Pull request by darth-tytus**
- Replace `RWOperation` with `map<string, string>`
- Update pre-commit hook to skip delated files

Resolve #1368

### CODE REVIEW

File: src/ReadStreamCSV.h [hunk 1]
```diff
@@ -228,16 +228,17 @@ class ReadStreamCSV : public ReadStream {
...
 class ReadFileCSV : public ReadStreamCSV {
 public:
-    ReadFileCSV(const RWOperation& rwOperation, SymbolTable& symbolTable, RecordTable& recordTable)
+    ReadFileCSV(const std::map<std::string, std::string>& rwOperation, SymbolTable& symbolTable,
+            RecordTable& recordTable)
             : ReadStreamCSV(fileHandle, rwOperation, symbolTable, recordTable),
               baseName(souffle::baseName(getFileName(rwOperation))),
               fileHandle(getFileName(rwOperation), std::ios::in | std::ios::binary) {
-        if (!rwOperation.has("intermediate")) {
+        if (rwOperation.find("intermediate") == rwOperation.end()) {
```
Comment (mmcgr) [inline src/ReadStreamCSV.h:236]: I think you can actually remove the if statement entirely and always execute the body. Loading of intermediate relations for incremental execution was removed a while ago.

File: src/WriteStream.h [hunk 2]
```diff
@@ -14,10 +14,10 @@
 
 #pragma once
 
-#include "RWOperation.h"
 #include "RamTypes.h"
 #include "RecordTable.h"
 #include "SymbolTable.h"
+#include "Util.h"
```
Comment (mmcgr) [inline src/WriteStream.h:20]: What is Util.h used for? It includes most system library headers so it's a heavy dependency to have.
std::map is used now, though, and should be added.

File: utilities/git/hooks/pre-commit.sh [hunk 3]
```diff
@@ -20,7 +20,7 @@ unset GIT_LITERAL_PATHSPECS
 
 #Change the delimiter used by 'for' so we can handle spaces in names
 IFS=$'\n'
-for FILE in $(git diff --cached --name-only "*.cpp" "*.h")
+for FILE in $(git diff --diff-filter=d --cached --name-only "*.cpp" "*.h")
```
Comment (mmcgr) [inline utilities/git/hooks/pre-commit.sh:23]: This is an entirely reasonable change, but should have been in a separate PR.

Comment (darth-tytus) [reply utilities/git/hooks/pre-commit.sh:23]: I agree. Should I create a separate PR?

Comment (mmcgr) [reply utilities/git/hooks/pre-commit.sh:23]: Don't bother. Now that we've looked at it and discussed it there's no gain.

**Pull request merged**